### PR TITLE
Yield Aggregator UI

### PR DESCRIPTION
**Pull request by chriswudev**
Main task: https://github.com/ubiquity/uad-ui/issues/13

Closes:
- https://github.com/ubiquity/uad-ui/issues/15
- https://github.com/ubiquity/uad-ui/issues/18
- https://github.com/ubiquity/uad-ui/issues/27
- https://github.com/ubiquity/uad-ui/issues/23
- https://github.com/ubiquity/uad-ui/issues/21
- https://github.com/ubiquity/uad-ui/issues/22
- https://github.com/ubiquity/uad-ui/issues/20
- ubiquity/ubiquity-dollar-development#119
- https://github.com/ubiquity/uad-ui/issues/28

### CODE REVIEW

**Review by Zequez — APPROVED** [review]

**Comment by chriswudev** [comment]
image

**Comment by 0x4007** [comment]
#### 📝 Changed routes:


| [**/styles/tailwind**](https://uad-ui-4mhd97cre-pavlovcik.vercel.app/styles/tailwind) |
|:-:|
|<a href="https://uad-ui-4mhd97cre-pavlovcik.vercel.app/styles/tailwind">Screenshot of /styles/tailwind</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/kZ-tWrUm5l4zupFzcQtOKy1_-7wFD39TCreT-PByTPX9s5ItQ3433FmsqqadHojwvGsMzomFHHT26adapEVf_A.png">(view full size)</a> |







Commit c00d5a0a6eb8e0c4186bfd1c00d9a72acec4c151 (https://uad-ui-4mhd97cre-pavlovcik.vercel.app).

**Review by Zequez — COMMENTED** [review]
Awesome job @chriswudev! It looks amazing!

**Comment by Zequez** [comment]
Captura de Tela 2021-10-01 às 04 32 55

Changed some of the spacing, fixed some typos and changed around some wording. So it looks like this now.

File: components/YieldFarming.tsx [hunk 1]
```diff
@@ -121,11 +136,11 @@ export const YieldFarmingSubcontainer = ({ actions, yieldProxyData, depositInfo,
             maxYieldBonusPct={yieldProxyData.bonusYieldMaxPct}
             baseDepositFeePct={yieldProxyData.depositFeeBasePct}
             minDepositFeePct={0}
-            balance={{ usdc: 200, ubq: 150, uad: 300 }}
+            balance={balance}
             onDeposit={actions.onDeposit}
             disable={isProcessing}
           />
-        )
+        ) : null
```
Comment (Zequez) [inline components/YieldFarming.tsx]: Would be nice to show a loading indicator on this case too.

File: components/YieldFarming.tsx [hunk 2]
```diff
@@ -24,6 +27,17 @@ export const YieldFarmingContainer = ({ contracts, account, signer }: UserContex
     setDepositInfo(di);
   }
 
+  useEffect(() => {
+    if (balances && contracts && account) {
+      contracts.usdc.balanceOf(account.address).then((bigUsdc) => {
```
Comment (Zequez) [inline components/YieldFarming.tsx]: I mean, I would have modified the context.balances object, but I guess this work too. It's alright we'll take care of it here https://github.com/pavlovcik/uad-ui/issues/29

**Review by Zequez — APPROVED** [review]

**Review by Zequez — APPROVED** [review]

**Comment by Zequez** [comment]
So what's left is for the DebtClock to be implemented on the YieldProxy contract for the uAR calculations and then update the uAR calculations here to take that into account too.

ubiquity/ubiquity-dollar-development#115
ubiquity/ubiquity-dollar-development#116

**Comment by Zequez** [comment]
And it would be nice to have this implemented so that we can all see the Vercel results, otherwise the checks fails and we can't see why. https://github.com/pavlovcik/uad-ui/issues/11

**Comment by 0x4007** [comment]
```
yarn run v1.22.15
$ run-s solidity:build next:build
$ ./build-scripts/compile-artifacts.sh
Submodule path 'contracts': checked out 'fffc20f7432da7dee273a11900e45c30fb89cf65'
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "hardhat-gas-reporter > eth-gas-reporter@0.2.22" has unmet peer dependency "@codechecks/client@^0.1.0".
[4/4] 🔨  Building fresh packages...
warning Ignored scripts due to flag.
$ TS_NODE_TRANSPILE_ONLY=1 yarn hardhat compile
$ ~/repos/ubiquity/uad/ui/contracts/node_modules/.bin/hardhat compile
Nothing to compile
Creating Typechain artifacts in directory artifacts/types for target ethers-v5
Successfully generated Typechain artifacts!
$ ~/repos/ubiquity/uad/ui/contracts/node_modules/.bin/hardhat export --export ../fixtures/full-deployment.json --network mainnet
$ next build
info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
Failed to compile.

./components/bonding.migrate.tsx:6:10
Type error: Module '"./context/connected"' declares 'Balances' locally, but it is not exported.

  4 | import { UbiquityAlgorithmicDollarManager } from "../contracts/artifacts/types/UbiquityAlgorithmicDollarManager";
  5 | import { EthAccount } from "./common/types";
> 6 | import { Balances, useConnectedContext } from "./context/connected";
    |          ^
  7 |
  8 | async function _migrate(
  9 |   provider: ethers.providers.Web3Provider | null,
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "next:build" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by 0x4007** [comment]
The boost amounts don't appear to be correct. These should be read dynamically from the contract.

https://github.com/ubiquity/uad-contracts/blob/fffc20f7432da7dee273a11900e45c30fb89cf65/contracts/YieldProxy.sol#L27-L30

```solidity
    uint256 public constant BONUS_YIELD_MAX = 10000; // 1000 = 10% 100 = 1% 10 = 0.1% 1 = 0.01%
    uint256 public bonusYield; //  5000 = 50% 100 = 1% 10 = 0.1% 1 = 0.01%
    uint256 public constant FEES_MAX = 100000; // 1000 = 1% 100 = 0.1% 10 = 0.01% 1 = 0.001%
    uint256 public constant UBQ_RATE_MAX = 10000e18; // 100000e18 Amount of UBQ to be stake to reduce the deposit fees by 100%
```

**Comment by 0x4007** [comment]
loops this ad infinitum `run-s solidity:build next:build`

```

00:54:10.901 | Cloning github.com/pavlovcik/uad-ui (Branch: feat/yield-aggregator-ui, Commit: 95868ff)
-- | --
00:54:14.378 | Warning: Failed to fetch one or more git submodules
00:54:14.379 | Cloning completed: 3.478s
00:54:14.416 | Analyzing source code...
00:54:15.955 | Installing build runtime...
00:54:18.228 | Build runtime installed: 2.273s
00:54:21.075 | Looking up build cache...
00:54:24.214 | Build cache downloaded [41.22 MB]: 2910.488ms
00:54:25.587 | Installing dependencies...
00:54:25.898 | yarn install v1.22.17
00:54:25.986 | [1/4] Resolving packages...
00:54:26.415 | [2/4] Fetching packages...
00:54:47.090 | [3/4] Linking dependencies...
00:54:56.680 | [4/4] Building fresh packages...
00:54:57.139 | $ husky install
00:54:57.237 | husky - Git hooks installed
00:54:57.242 | Done in 31.35s.
00:54:57.269 | Detected Next.js version: 11.0.0
00:54:57.271 | Running "yarn run build"
00:54:57.624 | yarn run v1.22.17
00:54:57.664 | $ run-s solidity:build next:build
00:54:58.227 | $ ./build-scripts/compile-artifacts.sh
00:54:59.170 | [1/4] Resolving packages...
00:54:59.603 | [2/4] Fetching packages...
00:55:00.130 | [3/4] Linking dependencies...
00:55:03.235 | [4/4] Building fresh packages...
00:55:03.690 | $ run-s solidity:build next:build
00:55:04.274 | $ ./build-scripts/compile-artifacts.sh
00:55:05.327 | [1/4] Resolving packages...
00:55:05.750 | success Already up-to-date.
00:55:06.047 | $ run-s solidity:build next:build
00:55:06.499 | $ ./build-scripts/compile-artifacts.sh
00:55:07.354 | [1/4] Resolving packages...
00:55:07.770 | success Already up-to-date.
00:55:08.057 | $ run-s solidity:build next:build
00:55:08.501 | $ ./build-scripts/compile-artifacts.sh
00:55:09.366 | [1/4] Resolving packages...
00:55:09.796 | success Already up-to-date.
00:55:10.076 | $ run-s solidity:build next:build
00:55:10.530 | $ ./build-scripts/compile-artifacts.sh

```

**Comment by Zequez** [comment]
There @pavlovcik it's ready for merging, all working.

**Review by Zequez — APPROVED** [review]
I still believe all the un-fancy CSS and the active video that auto-plays on the background are horrible mistakes, it makes even decent computer crawl to a halt and makes it so you can't keep the tab open unless you enjoy the sound of your CPU running at 100% and your computer being sluggish. I have to disable it all on local development because it makes it unbearable; at first I thought it was the codebase so big that VSCode would work slowly, but it isn't, it's the CSS of the app.

We should set a sort of performance target for the app, because this is just a terrible experience.

Aside from that, everything looks good.

**Comment by 0x4007** [comment]
I'm running the UI on a 6K display off my MacBook Air it's fine

**Comment by 0x4007** [comment]
@Zequez idea: lets make a lofi/hifi switch at the top of the UI where we can disable the blur/video/effects? We just add a class on the body and can inherit in the css!

**Comment by Zequez** [comment]
It's ready for merging, so let's just focus on the UI performance as a separate issue.

File: components/common/contracts-shortcuts.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  console.log("Actual UBQ per block", actualUbqPerBlock);
+  console.log("Extra UBQ per block to treasury", actualUbqPerBlock / ugovDivider);
+  const blockCountInAWeek = toNum(await contracts.bonding.blockCountInAWeek());
+  console.log("Block count in a week", blockCountInAWeek);
+
+  const ubqPerWeek = actualUbqPerBlock * blockCountInAWeek;
+  console.log("UBQ Minted per week", ubqPerWeek);
+  console.log("Extra UBQ minted per week to treasury", ubqPerWeek / ugovDivider);
+
+  const DAYS_IN_A_YEAR = 365.2422;
```
Comment (0x4007) [inline components/common/contracts-shortcuts.ts:220]: I'm impressed with this precision

File: components/common/contracts-shortcuts.ts [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+  const usdPerYear = usdPerDay * DAYS_IN_A_YEAR;
+  console.log("USD Minted per day", usdPerDay);
+  console.log("USD Minted per week", usdPerWeek);
+  console.log("USD Minted per year", usdPerYear);
+  const usdAsLp = 0.75;
+  const bigNumberOneUsdAsLp = ethers.utils.parseEther(usdAsLp.toString());
+
+  const bondingDiscountMultiplier = await contracts.bonding.bondingDiscountMultiplier();
+  const sharesResults = await Promise.all(
+    [1, 50, 100, 208].map(async (i) => {
```
Comment (0x4007) [inline components/common/contracts-shortcuts.ts:234]: What's up with these values? There should be people with different amounts in the future @Zequez @chriswudev

File: components/common/utils.ts [hunk 5]
```diff
@@ -41,3 +41,34 @@ export async function erc1155BalanceOf(addr: string, erc1155UbiquityCtr: ERC1155
...
   return fullBalance;
 }
+
+export const constrainNumber = (num: number, min: number, max: number): number => {
+  if (num < min) return min;
+  else if (num > max) return max;
+  else return num;
+};
+
+export const getNetworkName = (provider: ethers.providers.Web3Provider): string => {
```
Comment (0x4007) [inline components/common/utils.ts:51]: I feel like this would have been better implemented as a hashmap

File: components/uar.balance.tsx [hunk 6]
```diff
@@ -37,7 +40,7 @@ const UarBalance = () => {
   return (
     <>
       <div id="uar-balance">
-        <a target="_blank" href={`https://etherscan.io/token/${TOKEN_ADDR}${account ? `?a=${account.address}` : ""}`}>
+        <a target="_blank">
           <div>
             <span>
               <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 75 85.45">
```
Comment (0x4007) [inline components/uar.balance.tsx:46]: Perhaps we should convert the icons in the inventory to reused TSX variables

File: fixtures/named-accounts.json [hunk 7]
```diff
@@ -9,5 +9,7 @@
   "curveWhaleAddress": "0xC2872ab688400940d5a6041599b3F7de20730d49",
   "daiWhaleAddress": "0x16463c0fdB6BA9618909F5b120ea1581618C1b9E",
   "sushiMultiSig": "0x9a8541Ddf3a932a9A922B607e9CF7301f1d47bD1",
-  "ubq": "0xefC0e701A824943b469a694aC564Aa1efF7Ab7dd"
-}
+  "ubq": "0xefC0e701A824943b469a694aC564Aa1efF7Ab7dd",
+  "jarUSDCAddr": "0xEB801AB73E9A2A482aA48CaCA13B1954028F4c94",
```
Comment (0x4007) [inline fixtures/named-accounts.json:13]: For future reference, it is redundant to indicate that they are addresses in the property names of this object. I'm too lazy to change right now though lol

**Review by 0x4007 — APPROVED** [review]

File: components/common/contracts-shortcuts.ts (hunk 4 above)
Comment (Zequez) [reply components/common/contracts-shortcuts.ts:234]: This is just to show the APY at different weeks; it's not part of the UI just part of the `logBondingUbqInfo` function for the sake of developing and debugging.

**Pull request merged**